src/clients/AcrossSpokePoolClient.ts

```diff
@@ -0,0 +1,681 @@
+
+    // Sort all events to ensure they are stored in a consistent order.
+    events.forEach((events: Event[]) => sortEventsAscendingInPlace(events));
+
+    return {
+      success: true,
+      currentTime: currentTime.toNumber(), // uint32
+      firstDepositId,
+      latestBlockNumber,
+      latestDepositId,
+      searchEndBlock: searchConfig.toBlock,
+      events,
+    };
+  }
+
+  async update(eventsToQuery = this.queryableEventNames): Promise<void> {
+    if (this.configStoreClient !== null && !this.configStoreClient.isUpdated) {
+      throw new Error("RateModel not updated");
+    }
+
+    const { events: queryResults, currentTime, ...update } = await this._update(eventsToQuery);
+    if (!update.success) {
+      // This failure only occurs if the RPC searchConfig is miscomputed, and has only been seen in the hardhat test
+      // environment. Normal failures will throw instead. This is therefore an unfortunate workaround until we can
+      // understand why we see this in test. @todo: Resolve.
+      return;
+    }
+
+    if (eventsToQuery.includes("TokensBridged")) {
+      for (const event of queryResults[eventsToQuery.indexOf("TokensBridged")]) {
+        this.tokensBridged.push(spreadEventWithBlockNumber(event) as TokensBridged);
+      }
+    }
+
+    // For each depositEvent, compute the realizedLpFeePct. Note this means that we are only finding this value on the
+    // new deposits that were found in the searchConfig (new from the previous run). This is important as this operation
+    // is heavy as there is a fair bit of block number lookups that need to happen. Note this call REQUIRES that the
+    // hubPoolClient is updated on the first before this call as this needed the the L1 token mapping to each L2 token.
+    if (eventsToQuery.includes("FundsDeposited")) {
+      const allDeposits = [
+        ...(queryResults[eventsToQuery.indexOf("FundsDeposited")] as FundsDepositedEvent[]),
+        ...this.earlyDeposits,
+      ];
+      const { earlyDeposits = [], depositEvents = [] } = groupBy(allDeposits, (depositEvent) => {
+        if (depositEvent.args.quoteTimestamp > currentTime) {
+          const { args, transactionHash } = depositEvent;
+          this.logger.debug({
+            at: "SpokePoolClient#update",
+            message: "Deferring early deposit event.",
+            currentTime,
+            deposit: { args, transactionHash },
+          });
+          return "earlyDeposits";
+        } else {
+          return "depositEvents";
+        }
+      });
+      this.earlyDeposits = earlyDeposits;
+
+      if (depositEvents.length > 0) {
+        this.log("debug", `Fetching realizedLpFeePct for ${depositEvents.length} deposits on chain ${this.chainId}`, {
+          numDeposits: depositEvents.length,
+        });
+      }
+
+      const dataForQuoteTime: { realizedLpFeePct: BigNumber; quoteBlock: number }[] = await Promise.all(
```

Comment:
Replaced BlueBird `Promise.map(...)` with a `Promise.all({}.map(...))`

Reply:
Was concurrency being used in the bluebird promise?